language/return_spec.rb

```diff
@@ -249,4 +249,218 @@ def f
+            end
+
+            ScratchPad << "after begin"
+          END_OF_CODE
+
+          load @filename
+          ScratchPad.recorded.should == ["before begin"]
+        end
+
+        it "fires ensure block before returning" do
+          ruby_exe(<<-END_OF_CODE).should == "within ensure\n"
+            begin
+              return
+            ensure
+              puts "within ensure"
+            end
+
+            puts "after begin"
+          END_OF_CODE
+        end
+
+        it "does not fire ensure block before returning while loads file" do
+          File.write(@filename, <<-END_OF_CODE)
+            ScratchPad << "before begin"
+            begin
+              return
+            ensure
+              ScratchPad << "within ensure"
+            end
+
+            ScratchPad << "after begin"
+          END_OF_CODE
+
+          load @filename
+          ScratchPad.recorded.should == ["before begin"]
+        end
+
+        it "swallows exception if returns in ensure block" do
+          File.write(@filename, <<-END_OF_CODE)
+            begin
+              raise
+            ensure
+              ScratchPad << "before return"
+              return
+            end
+          END_OF_CODE
+
+          ->() {
+            load @filename
+          }.should_not raise_error
+
+          ScratchPad.recorded.should == ["before return"]
+        end
+      end
+
+      describe "within a block" do
+        it "is allowed" do
+          File.write(@filename, <<-END_OF_CODE)
+            ScratchPad << "before call"
+            proc { return }.call
+
+            ScratchPad << "after call"
+          END_OF_CODE
+
+          load @filename
+          ScratchPad.recorded.should == ["before call"]
+        end
+      end
+
+      describe "within a class" do
+        it "is allowed" do
+          File.write(@filename, <<-END_OF_CODE)
+            class A
+              ScratchPad << "before return"
+              return
+
+              ScratchPad << "after return"
+            end
+          END_OF_CODE
+
+          load @filename
+          ScratchPad.recorded.should == ["before return"]
+        end
+      end
+
+      describe "file loading" do
+        it "stops file loading and execution" do
+          File.write(@filename, <<-END_OF_CODE)
+            ScratchPad << "before return"
+            return
+            ScratchPad << "after return"
+          END_OF_CODE
+
+          load @filename
+          ScratchPad.recorded.should == ["before return"]
+        end
+      end
+
+      describe "file requiring" do
+        it "stops file loading and execution" do
+          File.write(@filename, <<-END_OF_CODE)
+            ScratchPad << "before return"
+            return
+            ScratchPad << "after return"
+          END_OF_CODE
+
+          require @filename
+          ScratchPad.recorded.should == ["before return"]
+        end
+      end
+
+      describe "return with argument" do
+        it "does not affect exit status" do
```

Comment:
This seems also a bug, filed as https://bugs.ruby-lang.org/issues/14062.
It seems OK to keep the spec like this for now though, could you just add a comment with the bug URL above?